### PR TITLE
Tasks submission to queue database optimization

### PR DESCRIPTION
**Pull request by codtiger**
## Description
Changing the function ```queue_submit.py``` in ```storage_sqlalchemy.py``` file, which does the query once on all the ids of items to be added to find the duplicates. Then adds all the non-duplicate ones. Also a benchmark file is also provided for adding a number of tasks raising from 1 to 1000 to measure how much of speedup the change gives. The results are brought in the file below.(The best run is selected for both the main repo and this repo's code.)
[runs_submit.txt](https://github.com/MolSSI/QCFractal/files/4385357/runs_submit.txt)

## Changelog description
Optimized the ```queue_submit``` function for task submission to queue in the database.

## Status

- [X] Code base linted
- [X] Ready to go

### CODE REVIEW

**Comment by codtiger** [comment]
**Note:** Based on this comment found in the original description of the function
`Tasks are unique by their base_result, which should be inserted into
        the DB first before submitting it's corresponding task to the queue
        (with result.status='INCOMPLETE' as the default)
        The default task.status is 'WAITING'`
and the fact that there are no cases of duplicate tasks(by base_result) insertion within the input ```data```  in the tests, duplicate cases within ```data``` was safely ignored. If this is to change, please let me know.

File: devtools/conda-envs/openff.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 name: qcarchive
 channels:
   - defaults
-  - psi4
+  - psi4/label/dev
```
Comment (doaa-altarawy) [inline devtools/conda-envs/openff.yaml:4]: @bennybp ?

File: qcfractal/storage_sockets/sqlalchemy_socket.py [hunk 2]
```diff
@@ -1897,33 +1897,53 @@ def queue_submit(self, data: List[TaskRecord]):
...
-                    self.logger.warning("queue_submit got a duplicate task: {}".format(task.to_dict()))
-                    results.append(str(task.id))
-                    meta["duplicates"].append(task_num)
-                # except Exception as err:
-                #     self.logger.warning('queue_submit submission error: {}'.format(str(err)))
-                #     meta["success"] = False
-                #     meta["errors"].append(str(err))
-                #     results.append(None)
+                    new_idx.append(task_num)
+                    task.priority = task.priority.value
```
Comment (doaa-altarawy) [inline qcfractal/storage_sockets/sqlalchemy_socket.py:1929]: Add the comment
# Must be an integer for sorting

Comment (doaa-altarawy) [reply qcfractal/storage_sockets/sqlalchemy_socket.py:1929]: @codtiger

Comment (codtiger) [reply qcfractal/storage_sockets/sqlalchemy_socket.py:1929]: Done in the new commit

**Pull request merged**